packages/delisp-core/src/syntax-utils.ts

```diff
@@ -154,6 +156,106 @@ function syntaxChildren<I>(s: Syntax<I>): Array<Expression<I>> {
   }
 }
 
+function moduleChildren<I>(m: Module<I>): Array<Syntax<I>> {
+  return m.body;
+}
+
+function expressionBindings<I>(e: Expression<I>): Identifier[] {
+  switch (e.node.tag) {
+    case "function":
+      return e.node.lambdaList.positionalArgs;
+    case "let-bindings":
+      return e.node.bindings.map(b => b.variable);
+    default:
+      return [];
+  }
+}
+
+function syntaxBindings<I>(s: Syntax<I>): Identifier[] {
+  if (isExpression(s)) {
+    return expressionBindings(s);
+  } else {
+    switch (s.node.tag) {
+      case "definition":
+      case "export":
+      case "type-alias":
```

Comment:
I get the rule is `unused-variables`, but maybe it would make sense to extend it to unused types as well?